src/entity-manager.ts

```diff
@@ -279,6 +293,16 @@ export class EntityManager {
     } else {
       throw `Tried to remove absent component ${def.name} from entity ${id}`;
     }
+
+    // update query cache
+    const systems = this._componentToSystems.get(def.name);
+    for (let name of systems ?? []) {
+      const es = this._systemsToEntities.get(name);
+      if (es) {
+        const indx = es.findIndex((v) => v === id);
```

Comment:
this is a linear scan and might be inefficient for large systems. We'd need to change _systemsToEntities to be Map<string, Set<number>> to fix this I think